src/services/memory.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::Context;
+use std::task::Poll;
+
+use anyhow::anyhow;
+
+use async_trait::async_trait;
+use bytes::Bytes;
+use futures::TryStreamExt;
+
+use crate::error::Error;
+use crate::error::Kind;
+use crate::error::Result;
+use crate::ops::OpRead;
+use crate::Accessor;
+use crate::BoxedAsyncReader;
+
+#[derive(Default)]
+pub struct Builder {
+    data: HashMap<String, Bytes>,
+}
+
+impl Builder {
+    pub fn add_bytes(&mut self, key: &str, data: Bytes) -> &mut Self {
```

Comment:
How about implementing `write` instead?

Reply:
That might be a feature, but I don't need it for now on.

Reply:
Ok, we can implement in the future.